MobileCenter/MobileCenter/Internals/Channel/MSLogManagerDefault.m

```diff
@@ -104,6 +78,35 @@ - (void)enumerateDelegatesForSelector:(SEL)selector withBlock:(void (^)(id<MSLog
   }
 }
 
+#pragma mark - Channel Delegate
+
+- (void)channel:(id<MSChannel>)channel willSendLog:(id<MSLog>)log {
+  [self enumerateDelegatesForSelector:@selector(willSendLog:)
+                            withBlock:^(id<MSLogManagerDelegate> delegate) {
+                              if ([[delegate groupId] isEqualToString:[channel.configuration groupId]]) {
+                                [delegate willSendLog:log];
```

Comment:
This methods are optional so I would say you need to check if the delegate responds to the selector first.